### PR TITLE
Add HTML pages for OIDC resources under /profile/

### PR DESCRIPTION
**Pull request by segiddins**
Part of https://github.com/rubygems/rubygems.org/issues/4126

Available on https://oidc-api-token.rubygems.org

Demo walkthrough: https://share.cleanshot.com/2Qx4jGC5

### CODE REVIEW

**Comment by segiddins** [comment]
From review w/ @indirect:

- [ ] soft delete for API key roles
- [ ] if generating name for form, generate unique name
- [ ] change yaml name to `push_gem.yml`
- [ ] add name to generated yaml
- [ ] on gh workflow page, add instructions: "bump version, push tag" -- link to bundler guide for "creating gem > releasing the gem" for gem bump info, then `rake release:source_control_push`
- [ ] change on push to '[v*]`
- [ ] in generated YAML, put server if not `https://rubygem.org`
- [ ] gem await use `--source` if not default
- [ ] `link_to_unless_current` on table partials

File: app/assets/javascripts/oidc_api_key_role_form.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  //       name: "api_key[rubygem_id]",
+  //       value: "",
+  //     })
+  //     .appendTo(".t-body form .api_key_rubygem_id_form");
+  // }
+
+  // function removeHiddenRubygemField() {
+  //   $("#" + hiddenRubygemId + ":hidden").remove();
+  // }
+});
```
Comment (indirect) [inline app/assets/javascripts/oidc_api_key_role_form.js:71]:
```suggestion
```

File: db/schema.rb [hunk 2]
```diff
@@ -194,6 +194,19 @@
...
+    t.string "uri", null: false
+    t.datetime "last_verified_at"
+    t.datetime "last_failure_at"
+    t.integer "failures_since_last_verification", default: 0
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.index ["linkable_id", "linkable_type", "uri"], name: "index_link_verifications_on_linkable_and_uri"
+    t.index ["linkable_type", "linkable_id"], name: "index_link_verifications_on_linkable"
+  end
+
```
Comment (indirect) [inline db/schema.rb:209]:
```suggestion
```

**Review by indirect — APPROVED** [review]
looks good! I spotted a couple of bits I don't think you meant to include, but I think this is good to go.

**Pull request merged**